spec/virtual_total_spec.rb

```diff
@@ -435,37 +435,144 @@ def model_with_children(count)
       let(:author3) { Author.create }
       let(:author4) { Author.create.tap { |a| a.create_books(1, :published => true) } }
 
-      # NOTE: rails converts the nil to a 0
-      it "calculates sum with one off query" do
-        authors
+      describe ":sum" do
+        # NOTE: rails converts the nil to a 0
+        it "calculates sum with one off query" do
+          authors
 
-        expect do
-          expect(authors.map(&:sum_recently_published_books_rating)).to eq([6, 5, 0, 0])
-        end.to make_database_queries(:count => 4)
+          expect do
+            expect(authors.map(&:sum_recently_published_books_rating)).to eq([6, 5, 0, 0])
+          end.to make_database_queries(:count => 4)
+        end
+
+        it "calculates sum from preloaded association" do
+          authors.each { |a| a.recently_published_books.load }
+
+          expect do
+            expect(authors.map(&:sum_recently_published_books_rating)).to eq([6, 5, nil, nil])
+          end.to_not make_database_queries
+        end
+
+        it "calculates sum from attribute" do
+          authors
+          query = Author.select(:id, :sum_recently_published_books_rating).order(:id).load
+          expect do
+            expect(query.map(&:sum_recently_published_books_rating)).to eq([6, 5, 0, 0])
+          end.to make_database_queries(:count => 2)
```

Comment:
I expected this to be zero. Any idea why it is hitting the database here?

maybe it is a string symbol kind of thing when calling `has_attribute?`